presto-main/src/main/java/com/facebook/presto/operator/GroupByHash.java

```diff
@@ -64,4 +64,6 @@ static GroupByHash createGroupByHash(
     boolean contains(int position, Page page, int[] hashChannels);
 
     int putIfAbsent(int position, Page page);
+
+    int compare(int leftGroupId, int rightGroupId);
```

Comment:
Thought about this a bit more. Can we instead do all the sorting based on the hash code? Not only would that fix this issue (we can could `getHashCode` instead of `compare`), but it would also make the spilling code more generic, so that it worked with any type, rather than just ones that are orderable